### PR TITLE
fix: storybook webpack builder config

### PR DESCRIPTION
**Pull request by targeral**
# PR Details

fix storybook webpack builder config

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [x] I have updated changeset
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: packages/cli/plugin-storybook/src/features/utils/webpackConfig.ts [hunk 1]
```diff
@@ -160,11 +160,11 @@ export const getCustomWebpackConfigHandle = async ({
         ...(builderConfig as any).tools,
         // maybe buildConfiger is app-tools`s config
         webpack: [
-          (builderConfig as any).tools.webpack,
+          ...((builderConfig as any).tools.webpack ?? []),
```
Comment (chenjiahan) [inline packages/cli/plugin-storybook/src/features/utils/webpackConfig.ts:163]: 这里 tools.webpack 可能是数组，也可能是一个对象。要不直接用 mergeBuilderConfig 来合并？

https://modernjs.dev/builder/zh/api/builder-core.html#mergebuilderconfig

Comment (10Derozan) [reply packages/cli/plugin-storybook/src/features/utils/webpackConfig.ts:163]: 是对象的时候，.tools一定存在吗？遇到个问题报tools是undefined了

Comment (chenjiahan) [reply packages/cli/plugin-storybook/src/features/utils/webpackConfig.ts:163]: 不一定存在，这里直接用 mergeBuilderConfig 最好，不用做判断了

Comment (targeral) [reply packages/cli/plugin-storybook/src/features/utils/webpackConfig.ts]: done @chenjiahan

Comment (targeral) [reply packages/cli/plugin-storybook/src/features/utils/webpackConfig.ts]: 辛苦在 approve 一下

**Review by chenjiahan — APPROVED** [review]

File: packages/cli/plugin-storybook/src/features/utils/webpackConfig.ts (hunk 1 above)
Comment (chenjiahan) [reply packages/cli/plugin-storybook/src/features/utils/webpackConfig.ts]: done~

**Pull request merged**